addon-test-support/-private/setup-window-mock.js

```diff
@@ -7,4 +7,5 @@ import { reset } from 'ember-window-mock';
 //
 export default function setupWindowMock(hooks = self) {
   hooks.beforeEach(reset);
+  hooks.after(reset);
```

Comment:
I could swear that they wear, last time I checked. 😂

Pardon me. I'll update the PR.

Reply:
name | description
-- | --
before (function) | Runs before the first test.
beforeEach (function) | Runs before each test.
afterEach (function) | Runs after each test.
after (function) | Runs after the last test.

Source: https://api.qunitjs.com/QUnit/module

Seems like the `before` and `after` hooks get underway... I'll investigate.

Reply:
Ah, so the reason this was failing is that the `after` spy was missing:
https://github.com/kaliber5/ember-window-mock/pull/79/commits/f1407aab10bf5a68e143c8a24508f073f7c07f34#diff-c7912ef40c32d85f2e444b6b7c390e0eR11

Reply:
Ah, you're right, of course. I was probably confusing this with `ember-mocha`, which supports the hooks-thing as well, to allow to use common `setupSomething()` functions to work with Mocha as well, and this only supports the *Each hooks: https://github.com/emberjs/ember-mocha/blob/master/addon-test-support/ember-mocha/setup-test.js#L70-L77

Any reason not to use `afterEach`?

Reply:
> Ah, you're right, of course. I was probably confusing this with `ember-mocha`, which supports the hooks-thing as well, to allow to use common `setupSomething()` functions to work with Mocha as well, and this only supports the *Each hooks: https://github.com/emberjs/ember-mocha/blob/master/addon-test-support/ember-mocha/setup-test.js#L70-L77

I've opened https://github.com/emberjs/ember-mocha/issues/266 to track the addition of the `before` and `after` hooks, which actually are supported by `mocha`, but just not added to `ember-mocha`.

> Any reason not to use `afterEach`?

Purely for performance reasons. Since we already `reset` before each test, we don't need to `reset` after each test, but only after the last test, so that the next module can start with a clean slate.

Alternatively we could use `before` and `afterEach`, which would have the same effect.

Reply:
Based on the discussion in https://github.com/emberjs/ember-mocha/issues/266#issuecomment-435835639, I would change this to _just_ `afterEach`, instead of `beforeEach` and `after`, if that's alright with you.

Reply:
yes, makes sense. 

I am trying to think of any edge case where this could make a difference (i.e. would be a breaking change), but hopefully that's shouldn't be the case!?

Reply:
This _could_ cause tests to fail, if you are mutating the `window` mock in a preceding module that _does not_ use `setupWindowMock`. Because then in the next module (that _does_ use `setupWindowMock`) the first test could fail, because the state would not have been reset until that very test has been run.

However, this is then indicative of a broken / leaky test in the first place.

Would there be an easy way to check whether the `window` mock is in pristine condition? If so, we could add an assertion in `before` to make sure that the previous module did clean up after itself.